gptme/prompts.py

```diff
@@ -209,7 +212,43 @@ def prompt_tools(examples: bool = True) -> Generator[Message, None, None]:
     yield Message("system", prompt.strip() + "\n\n")
 
 
+def prompt_systeminfo() -> Generator[Message, None, None]:
+    """Generate the system information prompt."""
+    if platform.system() == "Linux":
+        os_info = get_system_distro()
+        os_version = platform.uname().release
+    elif platform.system() == "Windows":
+        os_info = "Windows"
+        os_version = platform.version()
+    elif platform.system() == "Darwin":
+        os_info = "macOS"
+        os_version = platform.mac_ver()[0]
+    else:
+        os_info = "unknown"
+        os_version = ""
+
+    prompt = f"## System Information\n\n**OS:** {os_info} {os_version}".strip()
+
+    yield Message(
+        "system",
+        prompt,
+    )
+
+
+def get_system_distro() -> str:
+    """Get the system distribution name."""
+    regex = re.compile(r"^NAME=\"?([^\"]+)\"?")
+    if os.path.exists("/etc/os-release"):
+        with open("/etc/os-release") as f:
+            for line in f:
+                matches = re.search(regex, line)
+                if matches:
+                    return matches.string[matches.start(1) : matches.end(1)]
+    return "Linux"
```

Comment:
We should get the distro version here instead of using `platform.uname().release`

Reply:
Will check into this. Also getting the same on my laptop, capturing the kernel version instead of the distro version.